include/VirtualFunctionTemplate.h

```diff
@@ -68,6 +85,19 @@
     pair.second = [this](auto... args) { return this->functionName<VTableFillerUserType>(args...); };                  \
   })
 
+/** Save the old vtable to be accessible by CALL_BASE_FUNCTION_TEMPLATE and overwrite it with the
+ *  new implementation.
+ */
+#define OVERRIDE_VIRTUAL_FUNCTION_TEMPLATE(functionName)                                                               \
+  do {                                                                                                                 \
+    /* Copy the old vtable into _base before filling it with the new implementation */                                 \
+    ChimeraTK::for_each(functionName##_base_vtable.table, [&](auto& pair) {                                            \
```

Comment:
I think this is not going to work in a chain, because the functionName##_base_vtable name is not unique per instance. If ExceptionDummy wants to call the Dummy-Implementation and Dummy wants to call the NumericAddressed implementation, you need three tables.

Reply:
The VTABLE should be called identical on all levels (i.e. do not add "_base" to the name in DEFINE_VIRTUAL_FUNCTION_OVERRIDE_VTABLE). This macro should receive a second argument with the name of the direct base class (or maybe auto-detect it with some fancy C++ magic, but it will go wrong in case of multiple inheritance). It can then get the vtable from the specified direct base class.

Reply:
Fixed